crates/sui-types/src/crypto.rs

```diff
@@ -435,6 +435,23 @@ where
     get_key_pair_from_rng(&mut OsRng)
 }
 
+/// Wrapper function to return SuiKeypair based on key scheme string
+pub fn random_key_pair_by_type(
+    key_scheme: Option<String>,
+) -> Result<(SuiAddress, SuiKeyPair), anyhow::Error> {
+    match key_scheme.as_deref() {
+        Some("secp256k1") => {
+            let (addr, key_pair): (_, Secp256k1KeyPair) = get_key_pair();
+            Ok((addr, SuiKeyPair::Secp256k1SuiKeyPair(key_pair)))
+        }
+        Some("ed25519") | None => {
```

Comment:
I'd still prefer avoiding `None` implying it's ed25519, we should either get rid on None in this or upcoming PR